api/tests/unit/domain/services/certification-cpf-service_test.js

```diff
@@ -249,76 +249,123 @@ describe('Unit | Service | Certification CPF service', function () {
-          it('should return a validation failure', async function () {
-            // given
-            const birthCountry = 'FRANCE';
-            const birthCity = 'MARSEILLE';
-            const birthPostalCode = '1234';
-            const birthINSEECode = '1234';
-
-            const marseilleCpf = domainBuilder.buildCertificationCpfCity({
-              postalCode: '1234',
-              INSEECode: '1234',
-              name: 'MARSEILLE',
-            });
-            certificationCpfCityRepository.findByINSEECode.withArgs({ INSEECode: '1234' }).resolves([marseilleCpf]);
-            certificationCpfCityRepository.findByPostalCode.withArgs({ postalCode: '1234' }).resolves([marseilleCpf]);
+          context('when birthcity is defined', function () {
+            it('should return a validation failure', async function () {
+              // given
+              const birthCountry = 'FRANCE';
+              const birthCity = 'MARSEILLE';
+              const birthPostalCode = '1234';
+              const birthINSEECode = '1234';
+
+              const marseilleCpf = domainBuilder.buildCertificationCpfCity({
+                postalCode: '1234',
+                INSEECode: '1234',
+                name: 'MARSEILLE',
+              });
+              certificationCpfCityRepository.findByINSEECode.withArgs({ INSEECode: '1234' }).resolves([marseilleCpf]);
+              certificationCpfCityRepository.findByPostalCode.withArgs({ postalCode: '1234' }).resolves([marseilleCpf]);
 
-            const certificationCPFCountry = domainBuilder.buildCertificationCpfCountry.FRANCE();
-            certificationCpfCountryRepository.getByMatcher
-              .withArgs({ matcher: 'ACEFNR' })
-              .resolves(certificationCPFCountry);
+              const certificationCPFCountry = domainBuilder.buildCertificationCpfCountry.FRANCE();
+              certificationCpfCountryRepository.getByMatcher
+                .withArgs({ matcher: 'ACEFNR' })
+                .resolves(certificationCPFCountry);
 
-            // when
-            const result = await getBirthInformation({
-              birthCountry,
-              birthCity,
-              birthPostalCode,
-              birthINSEECode,
-              certificationCpfCountryRepository,
-              certificationCpfCityRepository,
+              // when
+              const result = await getBirthInformation({
+                birthCountry,
+                birthCity,
+                birthPostalCode,
+                birthINSEECode,
+                certificationCpfCountryRepository,
+                certificationCpfCityRepository,
+              });
+
+              // then
+              expect(result).to.deep.equal(
+                new CpfBirthInformationValidation().failure({
+                  certificationCandidateError:
+                    CERTIFICATION_CANDIDATES_ERRORS.CANDIDATE_BIRTH_INSEE_CODE_OR_BIRTH_POSTAL_CODE_REQUIRED,
+                })
```

Comment:
```suggestion
CERTIFICATION_CANDIDATES_ERRORS.CANDIDATE_BIRTH_INSEE_CODE_OR_BIRTH_POSTAL_CODE_AND_BIRTH_CITY_REQUIRED,
```

(soit le code insee, soit postal et ville)